src/flow-timeout.c

```diff
@@ -90,6 +90,18 @@ static inline Packet *FlowForceReassemblyPseudoPacketSetup(Packet *p,
     p->flags |= PKT_HAS_FLOW;
     p->flags |= PKT_PSEUDO_STREAM_END;
 
+    if (f->vlan_id[0] > 0) {
```

Comment:
while vlan 0 is officially reserved, it seems it is actually used. So I don't think we should assume this fields value being 0 to mean there is no vlan.

Reply:
the value uses just 11 bits, so we could use one of the other bits to indicate the field has been set.

Reply:
That does not work as flow does not currently have vlan_idx.
But checking if the value is set ( "if vlan_id[0] " ) should still be sufficient as this also supports vlan 0, or?

Reply:
I think you need to add it or we could missed to set and log vlan 0 on packet. Or at reverse log a vlan 0 that is not real.